src/pages/workspaces/workspace/Workflows.js

```diff
@@ -237,9 +289,17 @@ const FindWorkflowModal = ajaxCaller(class FindWorkflowModal extends Component {
     const { namespace, name, snapshotId } = selectedWorkflow
 
     this.setState({ selectedWorkflow })
+
     try {
       const selectedWorkflowDetails = await Methods.method(namespace, name, snapshotId).get()
-      this.setState({ selectedWorkflowDetails })
+
+      const methodAjax = Ajax().Methods.method(selectedWorkflow.namespace, selectedWorkflow.name, selectedWorkflow.snapshotId)
+
+      const config = _.maxBy('snapshotId', await methodAjax.configs())
+
+      this.setState({ selectedWorkflowDetails, methodAjax })
+
+      config ? this.setState({ workflowRename: config.name, editedWorkflowName: config.name, config }) : this.setState({ workflowRename: selectedWorkflow.name, editedWorkflowName: selectedWorkflow.name })
```

Comment:
For clarity, we could combine all these `setState` calls into one, and parameterize the data being set instead.